x-pack/metricbeat/module/aws/cloudwatch/metadata/ec2/ec2.go

```diff
@@ -37,6 +37,11 @@ func AddMetadata(logger *logp.Logger, regionName string, awsConfig awssdk.Config
 	monitoringStates := map[string]string{}
 	for instanceID, output := range instancesOutputs {
 		for eventIdentifier := range events {
+			// add host cpu/network/disk fields and host.id and rate metrics for all instances from both the monitoring
+			// account and linked source accounts if include_linked_accounts is set to true
+			addHostFields(events[eventIdentifier], instanceID)
```

Comment:
i think we need to compute `potentialInstanceID` first and pass it to this method instead of `instanceID`